pages/star-search/index.tsx

```diff
@@ -41,6 +51,28 @@ const SUGGESTIONS = [
 
 type SuggesionTypes = (typeof SUGGESTIONS)[number];
 
+async function updateComponentRegistry(name: string) {
```

Comment:
The amount of widgets dynamically loaded at the moment is low, one 😅 , but long term if the list of widgets explodes, an [LRU cache](https://redis.io/glossary/lru-cache/) might make sense especially in the context of mobile devices, but YAGNI for now.